pkg/tfgen/docs.go

```diff
@@ -1357,129 +1361,139 @@ func (g *Generator) convertExamplesInner(
 		contract.IgnoreError(err)
 	}
 
-	for _, section := range splitGroupLines(docs, "## ") {
-		if len(section) == 0 {
-			continue
+	// Find code fences in the presented doc
+	type tfBlockIndex struct {
+		start       int
+		end         int
+		headerStart int // a value of -1 indicates there's no header
+	}
+
+	var inCodeBlock bool
+	codeFence := "```"
+	h2 := "##"
+	h3 := "###"
+	var codeIndices []tfBlockIndex
+	var currentBlock tfBlockIndex
+	currentBlock.headerStart = -1
+
+	for i := 0; i < (len(docs) - len(codeFence)); i++ {
+
+		if inCodeBlock == true {
```

Comment:
I'd much rather that we represented `inCodeBlock` as a function call that parses to the end of a code block, rather then a variable here.

Then the logic would look something like this:

```go
for i := 0; i < (len(docs) - len(codeFence)); i++ {
    if block, ok := atCodeBlock(docs, i); ok {
        codeIndices = append(codeIndices, block)
        i = block.end + 1
        continue
    }
    if end, ok := atHeader(docs, i); ok {
        headerStart = i
        i = end+1
    }
}
```